py-polars/polars/utils/_construction.py

```diff
@@ -664,6 +665,11 @@ def _post_apply_columns(
     return pydf
 
 
+@functools.lru_cache(maxsize=128)
+def _is_polars_dtype_cached(dtype: Any, *, include_unknown: bool = False) -> bool:
+    return is_polars_dtype(dtype, include_unknown=include_unknown)
```

Comment:
If we keep a cache I don't see a point in not using the cached variant always. Let's just put the decorator on the `is_polars_type`.

Reply:
In other code paths, the argument passed as `dtype` may not be hashable. That causes the decorated function to fail. Do you want me to create a wrapper that handles non-hashable arguments gracefully and caches all calls with hashable arguments?